e2e-polybft/e2e/nexus_bridge_test.go

```diff
@@ -1891,3 +1891,196 @@ func convertToEthValues(sendAmount uint64) (uint64, *big.Int) {
 
 	return sendAmountDfm.Uint64(), ethgo.Ether(sendAmount)
 }
+
+func TestE2E_ApexBridgeWithNexus_BatchFailed_Test1(t *testing.T) {
+	const (
+		apiKey = "test_api_key"
+	)
+
+	ctx, cncl := context.WithCancel(context.Background())
+	defer cncl()
+
+	testMode := uint8(1)
+
+	apex := cardanofw.RunApexBridge(
+		t, ctx,
+		cardanofw.WithAPIKey(apiKey),
+		cardanofw.WithVectorEnabled(false),
+		cardanofw.WithNexusEnabled(true),
+		cardanofw.WithCustomConfigHandler(testMode),
+	)
+
+	userPrime := apex.CreateAndFundUser(t, ctx, uint64(500_000_000))
+	require.NotNil(t, userPrime)
+
+	txProviderPrime := apex.GetPrimeTxProvider()
+
+	startAmountNexus := uint64(1)
+	expectedAmountNexus := ethgo.Ether(startAmountNexus)
+
+	userNexus, err := apex.CreateAndFundNexusUser(ctx, startAmountNexus)
+	require.NoError(t, err)
+
+	err = cardanofw.WaitForEthAmount(ctx, apex.Nexus, userNexus, func(val *big.Int) bool {
+		return val.Cmp(expectedAmountNexus) == 0
+	}, 10, 10)
+	require.NoError(t, err)
+
+	fmt.Println("Nexus user created and funded")
+
+	receiverAddrNexus := userNexus.Address().String()
+	fmt.Printf("Nexus receiver Addr: %s\n", receiverAddrNexus)
+
+	t.Run("Test failed batch - test 1", func(t *testing.T) {
+		sendAmountDfm, _ := convertToEthValues(1)
+
+		ethBalanceBefore, err := cardanofw.GetEthAmount(ctx, apex.Nexus, userNexus)
+		fmt.Printf("ETH Amount before Tx %d\n", ethBalanceBefore)
+		require.NoError(t, err)
+
+		txHash, err := userPrime.BridgeNexusAmount(t, ctx, txProviderPrime, apex.Bridge.PrimeMultisigAddr,
```

Comment:
I think we should check if batch failed to execute. Probably we need something similar like `TestE2E_ApexBridge_BatchRecreated`